packages/@n8n/decorators/src/module/module.ts

```diff
@@ -16,13 +16,18 @@ export interface BaseEntity {
 	reload(): Promise<void>;
 }
 
-export interface TimestampedEntity {
+export interface TimestampedIdEntity {
 	id: string;
 	createdAt: Date;
 	updatedAt: Date;
 }
 
-export type EntityClass = new () => BaseEntity | TimestampedEntity;
+export interface TimestampedEntity {
```

Comment:
Not sure why you need that. BaseEntity already accepts not having id I think, and WithTimestamps extends BaseEntity

Reply:
I don't see it requiring an id 🤔 
```typescript
export interface BaseEntity {
	hasId(): boolean;
	save(options?: unknown): Promise<this>;
	remove(options?: unknown): Promise<this>;
	softRemove(options?: unknown): Promise<this>;
	recover(options?: unknown): Promise<this>;
	reload(): Promise<void>;
}
```

Reply:
So we need to change the interface, because we are not using the BaseEntity from typeorm in the module to define the entity, but a mixed in version of our own BaseEntity.